### ISSUE TITLE
Embedding dimension 0

### ISSUE THREAD

**Issue by DHAiRYA2048** [tiskw/patchcore-ad#4]
I am getting this error (shown in image) when I run main.py using `python main.py train -i data_train -o ./index.faiss`. 
some info about the dataset;

- train and test images stored in data_train and data_test folders repsectively
- RGB images with size 224x224

![image](https://github.com/tiskw/patchcore-ad/assets/66826063/0b699438-cebf-47b3-9c1d-617be038be21)

**Comment by DHAiRYA2048** [comment]
Here is the error which shows why embedding dimension is (0,). 
![image](https://github.com/tiskw/patchcore-ad/assets/66826063/9f9d83fb-f39c-44f9-9cd6-fc916aebc2c5)

Here is the part of code where the error is occurring. `embeddings_list` remains empty. Any idea why is this happening?
![image](https://github.com/tiskw/patchcore-ad/assets/66826063/a648e729-9462-49df-bc36-950e57384eff)

**Comment by tiskw** [comment]
Hi @DHAiRYA2048,

Thank you for your issue post with detailed information and sorry for my late response.

Honestly, I'm a bit confused because the 1st and 2nd images you've posted do not match (the feature extraction step finished but the feature shape is strange in the 1st image, however in the 2nd image, the script stopped by the error during the feature extraction step...).

However, one possibility is that the training images you've placed under `data_train/` are not recognized. For example, you can replicate the error of your 2nd image by the following:

```bash
# Create an empty directory
mkdir data_dummy

# Train without training images.
python3 main.py train -i data_dummy -o ./index.faiss
```

This is just my guess, but the suffix of your training image is neither ".jpg" nor ".png", isn't it? I'm sorry it is not written in the README, but the image files used for the training should have the suffix ".jpg" or ".png". This behavior comes from the following code:

[patchcore-ad/patchcore/dataset.py L.205](https://github.com/tiskw/patchcore-ad/blob/main/patchcore/dataset.py#L205)
```python
self.paths = [path for path in self.root.glob("**/*") if path.suffix in [".jpg", ".png"]]
```

If you want to use the other types of image files as training data, you have 2 choices:

* Please convert your training image to PNG images with the suffix ".png" or JPEG images with the suffix ".jpg".
* Please modify the script like the following. The following modification is an example to enable ".webp" files.

Before: [patchcore-ad/patchcore/dataset.py L.205](https://github.com/tiskw/patchcore-ad/blob/main/patchcore/dataset.py#L205)
```python
self.paths = [path for path in self.root.glob("**/*") if path.suffix in [".jpg", ".png"]]
```

After: [patchcore-ad/patchcore/dataset.py L.205](https://github.com/tiskw/patchcore-ad/blob/main/patchcore/dataset.py#L205)
```python
self.paths = [path for path in self.root.glob("**/*") if path.suffix in [".jpg", ".png", ".webp"]]
```

If my guess is not correct and your problem is not resolved, please let me know again with the more detailed information of your training data.

Regards,
Tetsuya

**Comment by DHAiRYA2048** [comment]
Thank you for your response. My files were in .bmp format, I didn't check the `dataset.py` file before running, my bad. Now it runs fine. If I face some other error, I will let you know in another issue.
